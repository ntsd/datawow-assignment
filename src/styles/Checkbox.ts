import styled from "@emotion/styled";

const CustomCheckbox = styled.div<{ checked: boolean }>`
  display: inline-block;
	flex-shrink: 0;
  width: 22px;
  height: 22px;
  background-color: ${({ checked }) => (checked ? "#585292" : "transparent")};
  border: 2px solid #585292;
  border-radius: 3px;
	cursor: pointer;
	color: ${({ checked }) => (checked ? "#fff" : "transparent")};
  font-weight: bold;
	line-height: 22px;
  text-align: center;
`;

export default CustomCheckbox;
