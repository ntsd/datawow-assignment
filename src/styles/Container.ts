import styled from "@emotion/styled";

const Container = styled.div`
  background-color: #f5f5f5;
  min-height: 100vh;

  @media (min-width: 1024px) {
    max-width: 768px;
    margin: 3rem auto;
    border-radius: 1rem;
    min-height: unset;
  }
`;

export default Container;
