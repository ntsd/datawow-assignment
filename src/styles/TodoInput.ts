import styled from "@emotion/styled";

const TodoInput = styled.input`
  width: 100%;
  border: none;

  &:focus {
    outline: none;
    box-shadow: none;
  }

  ::placeholder {
    color: #bcbcbc;
  }
`;

export default TodoInput;
