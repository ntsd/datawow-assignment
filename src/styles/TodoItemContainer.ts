import styled from "@emotion/styled";

const TodoItemContainer = styled.div`
  background-color: #fff;
  padding: 0.5rem 1rem;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 46px;
  border-radius: 99rem;
  column-gap: 1rem;
`;

export default TodoItemContainer;
