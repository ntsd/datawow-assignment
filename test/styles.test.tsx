import React from 'react'
import renderer from 'react-test-renderer'
import { expect, test } from 'vitest'
import Checkbox from '../src/styles/Checkbox'
import Container from '../src/styles/Container'
import TodoInput from '../src/styles/TodoInput'
import TodoItemContainer from '../src/styles/TodoItemContainer'

function toJson(component: renderer.ReactTestRenderer) {
	const result = component.toJSON()
	expect(result).toBeDefined()
	return result as renderer.ReactTestRendererJSON | renderer.ReactTestRendererJSON[]
}

const elements: { [key: string]: React.ReactElement } = {
	"Checkbox": <Checkbox checked={false}>X</Checkbox>,
	"Container": <Container>X</Container>,
	"TodoInput": <TodoInput></TodoInput>,
	"TodoItemContainer": <TodoItemContainer></TodoItemContainer>
}

Object.keys(elements).forEach(key => {
	test(`snapshot ${key}`, () => {
		const component = renderer.create(elements[key])
		const tree = toJson(component)
		expect(tree).toMatchSnapshot()
	})
})

