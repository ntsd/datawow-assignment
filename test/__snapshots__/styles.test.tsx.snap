// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot Checkbox 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1j4k4u4{display:inline-block;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;width:22px;height:22px;background-color:transparent;border:2px solid #585292;border-radius:3px;cursor:pointer;color:transparent;font-weight:bold;line-height:22px;text-align:center;}",
      }
    }
    data-emotion="css 1j4k4u4"
  />,
  <div
    checked={false}
    className="css-1j4k4u4"
  >
    X
  </div>,
]
`;

exports[`snapshot Container 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1cjri8q{background-color:#f5f5f5;min-height:100vh;}@media (min-width: 1024px){.css-1cjri8q{max-width:768px;margin:3rem auto;border-radius:1rem;min-height:unset;}}",
      }
    }
    data-emotion="css 1cjri8q"
  />,
  <div
    className="css-1cjri8q"
  >
    X
  </div>,
]
`;

exports[`snapshot TodoInput 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-gzr0dl{width:100%;border:none;}.css-gzr0dl:focus{outline:none;box-shadow:none;}.css-gzr0dl::-webkit-input-placeholder{color:#bcbcbc;}.css-gzr0dl::-moz-placeholder{color:#bcbcbc;}.css-gzr0dl:-ms-input-placeholder{color:#bcbcbc;}.css-gzr0dl::placeholder{color:#bcbcbc;}",
      }
    }
    data-emotion="css gzr0dl"
  />,
  <input
    className="css-gzr0dl"
  />,
]
`;

exports[`snapshot TodoItemContainer 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-11rgiw7{background-color:#fff;padding:0.5rem 1rem;box-sizing:border-box;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;justify-content:space-between;height:46px;border-radius:99rem;-webkit-column-gap:1rem;column-gap:1rem;}",
      }
    }
    data-emotion="css 11rgiw7"
  />,
  <div
    className="css-11rgiw7"
  />,
]
`;
